control-myplaces/src/main/java/fi/nls/oskari/control/myplaces/handler/MyPlacesLayersHandler.java

```diff
@@ -57,131 +60,97 @@ public void preProcess(ActionParameters params) throws ActionException {
+        final long id = ConversionHelper.getLong(params.getRequiredParam(PARAM_ID), -1);
+        if (!service.canModifyCategory(user, id)) {
+            throw new ActionDeniedException(
+                    "Tried to modify category: " + id);
         }
-        checkUserCanModifyCategories(user, categories);
-        long[] ids = categories.stream().mapToLong(MyPlaceCategory::getId).toArray();
         try {
-            LOG.debug("Updating Categories:", ids);
-            int updated = layerService.update(categories);
-            LOG.info("Updated", updated, "/", categories.size());
-        } catch (ServiceException e) {
-            LOG.warn(e);
-            throw new ActionException("Failed to update layers");
-        }
-        for (MyPlaceCategory layer : categories) {
+            final MyPlaceCategory category = layerService.getById(id)
+                    .orElseThrow(() -> new ActionParamsException("Couldn't find myplaces layer, " + id));
+            category.setName(params.getRequiredParam(PARAM_NAME));
+            category.getWFSLayerOptions()
+                .setDefaultFeatureStyle(JSONHelper.createJSONObject(params.getHttpParam(PARAM_STYLE)));
+            layerService.update(Collections.singletonList(category));
+            LOG.info("Updated category:", id);
             AuditLog.user(params.getClientIp(), params.getUser())
-                    .withParam("id", layer.getId())
-                    .withParam("name", layer.getCategory_name())
+                    .withParam("id", category.getId())
+                    .withParam("name", category.getName())
                     .updated(AuditLog.ResourceType.MYPLACES_LAYER);
-        }
-
-        ByteArrayOutputStream baos;
-        try {
-            baos = MyPlaceCategoryHelper.toGeoJSONFeatureCollection(categories, service.getBaseLayer());
-        } catch (IOException e) {
+            JSONObject response = toLayerJSON(category);
+            ResponseHelper.writeResponse(params, 200, response);
+        } catch (ServiceException e) {
             LOG.warn(e);
-            throw new ActionException("Failed to create response");
+            throw new ActionException("Failed to update layers");
         }
-
-        ResponseHelper.writeResponse(params, 200,
-                ResponseHelper.CONTENT_TYPE_JSON_UTF8, baos);
     }
 
     @Override
     public void handleDelete(ActionParameters params) throws ActionException {
         final User user = params.getUser();
-        final String layerIds = params.getRequiredParam(PARAM_LAYERS);
-        final long[] ids = Arrays.stream(layerIds.split(","))
-                .mapToLong(Long::parseLong)
-                .toArray();
-        for (long id : ids) {
-            if (!service.canModifyCategory(user, id)) {
-                throw new ActionDeniedException("Tried to delete category: " + id);
-            }
+        final long id = ConversionHelper.getLong(params.getRequiredParam(PARAM_ID), -1);
```

Comment:
Prefer `ActionParameters.getRequiredParamLong(PARAM_ID)` instead. If the `id` is not a `long` we don't want to fail with `ActionDeniedException` but with a `ActionParamsExcepton` instead.